packages/common/src/types/index.ts

```diff
@@ -73,26 +57,47 @@ export type Store = {
   name: string;
 };
 
+export interface InvoiceLine extends DomainObject {
+  id: string;
+
+  itemId: string;
+  itemName: string;
+  itemCode: string;
+  itemUnit: string;
+  packSize: number;
+  numberOfPacks: number;
+  costPricePerPack: number;
+  sellPricePerPack: number;
+
+  expiryDate?: string | null;
+
+  batch?: string | null;
+
+  stockLine?: StockLine | null;
+
+  location?: string;
+  comment?: string;
+}
+
 export interface Invoice extends DomainObject {
   id: string;
   color: string;
-  comment: string;
-  theirReference: string;
+  comment?: string | null;
+  theirReference?: string | null;
```

Comment:
undefined | null | string is a bit ugh

Reply:
yeah. needs both, I take it? can't just have `comment: string | null`?